kornia/testing/__init__.py

```diff
@@ -260,31 +272,27 @@ def assert_close(
         **kwargs: Any,
     ) -> None:
         if rtol is None and atol is None:
-            with contextlib.suppress(Exception):
-                rtol, atol = _get_default_tolerance(actual, expected)
-
-        return _assert_close(actual, expected, rtol=rtol, atol=atol, check_stride=False, equal_nan=True, **kwargs)
+            # `torch.testing.assert_close` used different default tolerances than `torch.testing.assert_allclose`.
+            # TODO: remove this special handling as soon as https://github.com/kornia/kornia/issues/1134 is resolved.
+            #  Basically, this whole wrapper function can be removed and `torch.testing.assert_close` can be used
+            #  directly.
+            rtol, atol = _default_tolerances(actual, expected)
+
+        return _assert_close(
+            actual,
+            expected,
+            rtol=rtol,
+            atol=atol,
+            # this is the default value for torch>=1.10, but not for torch==1.9
+            # TODO: remove this if kornia relies on torch>=1.10
+            check_stride=False,
+            equal_nan=True,
+            **kwargs,
+        )
 
 except ImportError:
-    # Partial backport of torch.testing.assert_close for torch<1.9
     # TODO: remove this branch if kornia relies on torch>=1.9
-    from torch.testing import assert_allclose as _assert_close
-
-    class UsageError(Exception):
-        pass
-
-    def assert_close(
-        actual: torch.Tensor,
-        expected: torch.Tensor,
-        *,
-        rtol: Optional[float] = None,
-        atol: Optional[float] = None,
-        **kwargs: Any,
-    ) -> None:
-        try:
-            return _assert_close(actual, expected, rtol=rtol, atol=atol, **kwargs)
-        except ValueError as error:
-            raise UsageError(str(error)) from error
+    from torch.testing import assert_allclose as assert_close
```

Comment:
The old "backport" only handles some exception type wrangling that is no longer needed.